lib/Dancer/Request.pm

```diff
@@ -99,6 +99,18 @@ sub is_delete             { $_[0]->{method} eq 'DELETE' }
 sub is_patch              { $_[0]->{method} eq 'PATCH' }
 sub header                { $_[0]->{headers}->header($_[1]) }
 
+# We used to store the whole raw unparsed body; this was a big problem for large
+# file uploads (Issue 1129).  This convenient accessor should suit the
+# requirements of anyone who was fetching the request body, without having file
+# uploads stored in RAM.
+sub body {
+    my $http_body = shift->{_http_body} or return '';
+    my $body_fh = $http_body->body or return '';
+    $body_fh->seek(0, 0);
```

Comment:
As the filehandle is virtual and deal with a memory structure, I expect that wouldn't affect performance much. So yeah, that doesn't look /too/ insane.